api/envoy/api/v2/route/route.proto

```diff
@@ -939,6 +939,22 @@ message HeaderMatcher {
     // If specified, header match will be be performed based on whether the header is in the
     // request.
     bool present_match = 7;
+
```

Comment:
Should we bring `StringMatch` here and potentially deprecate `regex_match` and `exact_match`?

Reply:
That would be optimal, though, we have gone through a lot of churn here and TBH changes in this structure are a huge PITA for all users, so I would probably say it's not worth it.

Reply:
I added a validation to check it's not empty, also updated the comment. Do we also want to check this in code and always return false if the value is empty?